src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonChare.hpp

```diff
@@ -87,14 +89,16 @@ struct WorldtubeSingleton {
         tmpl::list<Tags::EvolvedPosition<Dim>, Tags::EvolvedVelocity<Dim>>>;
   };
   using step_actions =
-      tmpl::list<Actions::ChangeSlabSize, Actions::ReceiveElementData,
+      tmpl::list<Actions::UpdateFunctionsOfTime, Actions::ChangeSlabSize,
+                 Actions::ReceiveElementData,
                  ::Actions::MutateApply<IterateAccelerationTerms>,
                  Actions::SendAccelerationTerms<Metavariables>,
                  ::Actions::MutateApply<UpdateAcceleration>,
                  ::Actions::RecordTimeStepperData<worldtube_system>,
                  ::Actions::UpdateU<worldtube_system>,
                  ::Actions::CleanHistory<worldtube_system, false>,
-                 Actions::SendToElements<Metavariables>>;
+                 Actions::SendToElements<Metavariables>,
+                 domain::Actions::CheckFunctionsOfTimeAreReady<Dim>>;
```

Comment:
Why check if the functions of time are ready at the end of the actions?

Reply:
They should be checked after `UpdateFunctionsOfTime` but before `AdvanceTime` so this is the most efficient place for it.

Reply:
Are the functions of time needed in any of the actions between `UpdateFunctionsOfTime` and `CheckFoTAreReady`? If so, how do you guarantee they are up to date?

Reply:
No, they are not needed. The singleton computes all the orbit parameters of the particle and then sets the functions of time accordingly. All the actions on the singleton simply use the orbit parameters directly rather than going through the functions of time.